### PR TITLE
feat: enabled lockdrop dispatch feature for Astar and Shiden

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* feat: enabled lockdrop dispatch feature for Astar and Shiden

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**
image

image

image

### CODE REVIEW

File: src/v2/services/implementations/MetamaskWalletService.ts [hunk 1]
```diff
@@ -126,18 +126,23 @@ export class MetamaskWalletService extends WalletService implements IWalletServi
...
-          lockdropDispatchAbi as AbiItem[],
-          evmPrecompiledContract.lockdropDispatch
-        );
+        // Todo: remove this logic once contract address for Shiden lockdrop has been updated to align with Astar and Shibuya
+        const chainId = await web3.eth.getChainId();
+        const isShiden = chainId === 336;
+
+        const contractAddress = isShiden
+          ? '0x0000000000000000000000000000000000005005'
+          : evmPrecompiledContract.lockdropDispatch;
```
Comment (gtg7784) [inline src/v2/services/implementations/MetamaskWalletService.ts:135]: if it need to be updated at every upgrades, I think we can create objects for each contract address mapped with chainId like this. 

```
evmPrecompiledContract.lockdropDispatch = {
336: '0x0000000000000000000000000000000000005005',
592: '0x0000000000000000000000000000000000005007',
81: '0x0000000000000000000000000000000000005007'
}
```

it's not necessary for now, just a comment.

**Review by gtg7784 — APPROVED** [review]
LGTM!

File: src/v2/services/implementations/MetamaskWalletService.ts (hunk 1 above)
Comment (impelcrypto) [reply src/v2/services/implementations/MetamaskWalletService.ts:140]: No, the contract address for Shiden (...5005) will be changed to 5007 soon, so we don't need to tweak the UI :D

Comment (gtg7784) [reply src/v2/services/implementations/MetamaskWalletService.ts:140]: Cool!

**Review by PierreOssun — APPROVED** [review]
Looks good!

**Review by bobo-k2 — APPROVED** [review]

**Pull request merged**